drivers/gpio/gpio_esp32.c

```diff
@@ -124,7 +93,7 @@ static void config_drive_strength(u32_t pin, int flags)
 {
 	volatile u32_t *reg = gpio_pin_reg(pin);
 
-	if ((flags & GPIO_DS_DISCONNECT_LOW) == GPIO_DS_DISCONNECT_LOW) {
+	if ((flags & GPIO_OPEN_SOURCE) == GPIO_OPEN_SOURCE) {
```

Comment:
According to the reference manual ESP32 platform supports GPIO open drain mode, not open source. This section should be written as
```
if (flags & GPIO_SINGLE_ENDED) {
	if (flags & GPIO_LINE_OPEN_DRAIN) {
		*reg |= GPIO_PIN_PAD_DRIVER;
	} else {
		r = -ENOTSUP;
	}
} else {
	*reg &= ~GPIO_PIN_PAD_DRIVER;
}
```
and placed in `gpio_esp32_config` function.